### PR TITLE
add validation for non-empty SB intermediate

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Resolves https://github.com/dotnet/source-build/issues/3337

Resulting exception:
```
./SourceBuildArcade.targets(83,5): error : No source-build artifacts were produced and the resulting intermediate is empty.
```

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets [hunk 1]
```diff
@@ -79,6 +79,10 @@
       <IntermediateNupkgArtifactFile Remove="$(CurrentRepoSourceBuildArtifactsPackagesDir)**\*.symbols.nupkg" />
     </ItemGroup>
 
+    <Error
+      Text="No source-build artifacts were produced and the resulting intermediate is empty."
```
[the remark this answers is not in the document]
Comment (oleksandr-didyk) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets:83]: Good point, will change, thanks

**Comment by oleksandr-didyk** [comment]
Test failure in the previous run was caused by a hardware issue on Helix's side (device used for tests was not accessible).
Will monitor the latest run and if anything will contact FR for help

**Review by MichaelSimons — APPROVED** [review]

**Comment by oleksandr-didyk** [comment]
Test fails consistently, reported the issue to the FR

**Comment by oleksandr-didyk** [comment]
@MichaelSimons changes from `main` fixed the CI issue this branch was having, PR is ready for merge after re-approval. Thanks!

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**